src/common/kafka/serializer/DA00HistogramSerializer.h

```diff
@@ -165,6 +188,12 @@ class HistogramSerializer : public AbstractSerializer {
     R MaxValue = XAxisValues.back();                           // maximum value
     R Step = (MaxValue - MinValue) / (XAxisValues.size() - 1); // step size
 
+    if (Time < MinValue) {
+      XTRACE(DATA, DEB, "Time %zi lower then min value %zi data dropped.", Time,
+             MinValue);
+      Stats.DataBeforeTimeOffsetDropped++;
+      return;
+    }
```

Comment:
I don't fully understand why someone would want to set a `BinOffset`, but under that assumption that there is a very good reason for this why do you _always_ drop values which have a binning-axis-value (time) less than the binned-minimum?
You have a `BinningStrategy` which can switch between dropping too-big binning-axis-values or adding their values to the last bin. Would it be worthwhile using the same mechanism to allow capturing below-minimum values too?

Reply:
I think I confused on this
> If you store only the upper edges or only the bin-centers you still need to make an assumption about something. If you store the bin edges there is no ambiguity at the cost of one number per message.

and overthinked. But I do not mind, now there is a possibility to offset the bins edges from 0 in case.